apps/api/src/app/integrations/e2e/create-integration.e2e.ts

```diff
@@ -95,6 +95,39 @@ describe('Create Integration - /integration (POST)', function () {
     expect(firstIntegration.active).to.equal(false);
     expect(secondIntegration.active).to.equal(true);
   });
+
+  it('should create custom SMTP integration with TLS options successfully', async function () {
+    const payload = {
+      providerId: 'nodemailer',
+      channel: 'email',
+      credentials: {
+        host: 'smtp.example.com',
+        port: '587',
+        secure: 'true',
+        requireTls: true,
+        tlsOptions: JSON.stringify({ rejectUnauthorized: false }),
```

Comment:
```suggestion
        tlsOptions: { rejectUnauthorized: false },
```
We are expecting to send an object for TLS options so the test should reflect that.
Also we need to review the API calls from the Web App, as it would need to transform from the string value stored in the form to the object we need to send. Same when retrieving the integration to convert from object to string to fill the form field when editing the integration. And for the update case.
https://github.com/novuhq/novu/blob/371c1381afb3e6b919a4355a649585880da65150/apps/web/src/api/integration.ts#L16

And not forgetting handle cases of malformed JSON. Not a straight forward implementation was this task. 🙁